src/reuse/project.py

```diff
@@ -137,30 +137,40 @@ def spdx_info_of(self, path: PathLike) -> SpdxInfo:
         """Return SPDX info of *path*.
 
         This function will return any SPDX information that it can find, both
-        from within the file and from the .reuse/dep5 file.
+        from within the file, the .license file and from the .reuse/dep5 file.
+
+        It also returns a single primary source path of the license/copyright
+        information, where 'primary' means '.license file' > 'header' > 'dep5'
```

Comment:
This doesn't really make sense to me. It should return the _correct_ source. If the returned `SpdxInfo` returns the aggregate of the file contents + the DEP5 file, then the source should be `'my_file'` for the lines extracted from the file, and `.reuse/dep5` for the lines extracted from DEP5.

This would change the return type of this function from `SpdxInfo` to `List[SpdxInfo]`. That's fine by me.

Furthermore, the source information should probably contain two grains of information:

- `source_type` which is an Enum à la `SourceType.FILE`, `SourceType.DOT_LICENSE`, `SourceType.DEP5` (and later `SourceType.REUSE_YAML`). This information is _programmatically_ useful.
- `source_path` which is a path to the exact file that contains the information. This information is _semantically_ useful.

If we want to implement that, we should probably change `SpdxInfo` into a dataclass first (and rename it to `ReuseInfo` while we're at it).

Reply:
Ah, if we don't aggregate, then there's no need to refactor, you're right.

I still think it makes sense to rename `license_path` to `source_path` and to also encode `source_type`.

Also to rename `SpdxInfo` to `ReuseInfo`, but we can put that off and open another issue for that.

Reply:
I'll resolve this as @carmenbianca already opened another issue for that.

Reply:
I encoded `source_type` not and refactored all the class and function names.